pkg/execute/kubectl.go

```diff
@@ -0,0 +1,206 @@
+package execute
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+
+	"github.com/kubeshop/botkube/pkg/config"
+	"github.com/kubeshop/botkube/pkg/utils"
+)
+
+const (
+	kubectlDisabledMsgFmt            = "Sorry, the admin hasn't given me the permission to execute kubectl command on cluster '%s'."
+	kubectlNotAuthorizedMsgFmt       = "Sorry, this channel is not authorized to execute kubectl command on cluster '%s'."
+	kubectlNotAllowedNamespaceMsgFmt = "Sorry, the kubectl command cannot be executed in the '%s' Namespace on cluster '%s'."
+	kubectlNotAllowedKindMsgFmt      = "Sorry, the kubectl command is not authorized to work with '%s' resources on cluster '%s'."
+	kubectlDefaultNamespace          = "default"
+)
+
+// Kubectl executes kubectl commands using local binary.
+type Kubectl struct {
+	log        logrus.FieldLogger
+	cfg        config.Config
+	resMapping ResourceMapping
+	runCmdFn   CommandRunnerFunc
+}
+
+// NewKubectl creates a new instance of Kubectl.
+func NewKubectl(log logrus.FieldLogger, cfg config.Config, mapping ResourceMapping, fn CommandRunnerFunc) *Kubectl {
+	return &Kubectl{
+		log:        log,
+		cfg:        cfg,
+		resMapping: mapping,
+		runCmdFn:   fn,
+	}
+}
+
+// CanHandle returns true if it's allowed kubectl command that can be handled by this executor.
+//
+// TODO: we should just introduce a command name explicitly. In this case `@BotKube kubectl get po` instead of `@BotKube get po`
+// As a result, we are able to detect kubectl command but say that you're simply not authorized to use it instead of "Command not supported. (..)"
+func (e *Kubectl) CanHandle(args []string) bool {
+	if len(args) == 0 {
+		return false
+	}
+
+	// Check if such kubectl verb is enabled
+	if !e.resMapping.AllowedKubectlVerbMap[args[0]] {
+		return false
+	}
+
+	return true
+}
+
+// Execute executes kubectl command based on a given args.
+//
+// This method should be called ONLY if:
+// - we are a target cluster,
+// - and Kubectl.CanHandle returned true.
+func (e *Kubectl) Execute(command string, isAuthChannel bool) (string, error) {
+	e.log.Infof("message %s, auth %v", command, isAuthChannel)
```

Comment:
```suggestion
	e.log.WithField("isAuthChannel", isAuthChannel).Infof("Handling message %q...", command)
```

Reply:
that was just for debugging, but I can leave it and apply your suggestion 👍